packages/block-library/src/post-author/edit.js

```diff
@@ -21,32 +19,39 @@ import {
 	withFontSizes,
 } from '@wordpress/block-editor';
 import { PanelBody, SelectControl, ToggleControl } from '@wordpress/components';
-import { useSelect } from '@wordpress/data';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 
 const DEFAULT_AVATAR_SIZE = 24;
 
-async function getAuthorDetails( authorId, setAuthorDetails ) {
-	const authorDetails = await apiFetch( {
-		path: '/wp/v2/users/' + authorId + '?context=edit',
-	} );
-	setAuthorDetails( authorDetails );
-}
-
-function PostAuthorDisplay( {
-	postAuthor,
-	setPostAuthor,
-	authorDetails,
-	setAuthorDetails,
-	props,
+function PostAuthorEdit( {
+	isSelected,
+	fontSize,
+	setFontSize,
+	context,
+	attributes,
+	setAttributes,
 } ) {
-	const ref = useRef();
+	const { postType, postId } = context;
+
+	const { authorId, authorDetails, authors } = useSelect( ( select ) => {
+		const { getEditedEntityRecord, getUser, getAuthors } = select( 'core' );
+		const _authorId = getEditedEntityRecord(
+			'postType',
+			postType,
+			postId
+		)?.author;
+
+		return {
+			authorId: _authorId,
+			authorDetails: _authorId ? getUser( _authorId ) : null,
+			authors: getAuthors(),
+		};
+	} );
```

Comment:
You're missing a dependency array here. I think the dependency array should be `[ postId, postType ]`.

Reply:
Good catch. I do this far too often 😞

Reply:
A custom ESLint rule might help.

Reply:
> A custom ESLint rule might help.

I believe it's what's meant to be proposed with #20936. The rule doesn't always neatly apply though. For example, the case described at https://github.com/WordPress/gutenberg/pull/22279#discussion_r423737059.